### PR TITLE
feat(cb2-3183) transform records from flat-tech-records dynamodb

### PR DESCRIPTION
**Pull request by aaron-downing**
## NOP receives Tech Record Data from new DB

[Previous work](https://dvsa.atlassian.net/browse/CB2-3462) on updating the `technical-records` DynamoDB data structure by creating a new DynamoDB (`flat-tech-records`) where the record is flat has meant that this lambda (cvs-tsk-edh-marshaller) is to be updated in order for the data from flat-tech-records to populate NOP.

A new environment variable (`PROCESS_FLAT_TECH_RECORDS`) has been created which controls what vehicle technical record DynamoDB (`technical-records` or `flat-tech-records`) populates NOP. 

The `cvs-edh-dispatcher-technical_records-${BRANCH}-queue` destination SQS queue will continue to be utilized for both `flat-tech-records` and `technical-records` streams so that there is no need for a new SQS queue to be created for streams originating from the `flat-tech-records` DynamoDB as it is intended that only one set of streams will populate this queue.

[link to ticket number](https://dvsa.atlassian.net/browse/CB2-3183)

## Checklist

- [x] Code has been tested manually
- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/functions/edh-marshaller.ts [hunk 1]
```diff
@@ -8,14 +8,17 @@ import { SQSService } from '../services/sqs';
...
 
 /**
- * λ function to process a DynamoDB stream of test results into a queue for certificate generation.
+ * λ function to process a DynamoDB stream and forward it to an appropriate SQS queue.
  * @param event - DynamoDB Stream event
  * @param context - λ Context
  * @param callback - callback function
  */
 const edhMarshaller: Handler = async (event: DynamoDBStreamEvent): Promise<void | Array<PromiseResult<SendMessageResult, AWSError>>> => {
+  const processFlatTechRecords = process.env.PROCESS_FLAT_TECH_RECORDS || 'false';
```
Comment (naathanbrown) [inline src/functions/edh-marshaller.ts]: If we are passing this env in does it not make more sense to make it a bool? Or even convert it to one?

Comment (aaron-downing) [reply src/functions/edh-marshaller.ts]: I was going to do that but environment variables are stored as strings and so you would have to check and then cast them to boolean which I thought would just be additional redundant code.

Comment (danielchildsbjss) [reply src/functions/edh-marshaller.ts]: How about 
`const processFlatTechRecords = process.env.PROCESS_FLAT_TECH_RECORDS === 'true';`

File: tests/unit/functions/edh-marshaller.test.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
-/* eslint-disable jest/no-conditional-expect */
 import { edhMarshaller } from '../../../src/functions/edh-marshaller';
 import { SQSService } from '../../../src/services/sqs';
 import { Context } from 'aws-lambda';
+import * as _ from '../../../src/utils/transformTechRecord';
```
Comment (danielchildsbjss) [inline tests/unit/functions/edh-marshaller.test.ts]: Is this used? If so why is it _?

File: tests/unit/functions/edh-marshaller.test.ts [hunk 3]
```diff
@@ -71,20 +182,21 @@ describe('edhMarshaller Function', () => {
         jest.spyOn(console, 'log');
 
         try {
```
Comment (danielchildsbjss) [inline tests/unit/functions/edh-marshaller.test.ts]: There are nicer ways to test exceptions being thrown
`expect(t).toThrow("Uh oh!");`

File: tests/unit/functions/edh-marshaller.test.ts (hunk 2 above)
Comment (aaron-downing) [reply tests/unit/functions/edh-marshaller.test.ts]: In order to use jest.spyOn you have to have the function be part of a module/class....
```
jest.spyOn(_, 'transformTechRecord').mockImplementation(transformRecordMock);
```

File: src/functions/edh-marshaller.ts (hunk 1 above)
Comment (aaron-downing) [reply src/functions/edh-marshaller.ts]: Created a new utilities function that handles it a bit more elegantly

File: src/utils/utilities.ts [hunk 4]
```diff
@@ -5,6 +5,16 @@ import { DynamoDBRecord } from 'aws-lambda';
 import { Configuration } from './config';
 import { ERROR } from '../models/enums';
 
+export const getTechRecordEnvVar = (): boolean => {
+  if(process.env.PROCESS_FLAT_TECH_RECORDS == 'false'){
```
Comment (premghinde) [inline src/utils/utilities.ts]: run linter

File: src/utils/transformTechRecord.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { DynamoDBRecord, AttributeValue } from 'aws-lambda';
```
Comment (premghinde) [inline src/utils/transformTechRecord.ts]: file names are lowercased and hyphenated

**Review by premghinde — DISMISSED** [review]

**Review by naathanbrown — DISMISSED** [review]

**Review by premghinde — DISMISSED** [review]

**Review by danielchildsbjss — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Comment by premghinde** [comment]
Approved

**Pull request merged**